packages/xo-web/src/common/xo/index.js

```diff
@@ -1207,8 +1207,33 @@ export const copyVms = (vms, type) => {
         _vms.map((vm, index) => cloneVm({ id: vm }, false, names[index]))
       )
     }
-
     if (sr !== undefined) {
+      if (process.env.XOA_PLAN < 3) {
+        const promise = [_call('xo.getAllObjects', { filter: { id: sr } })]
+        _vms.map(id =>
+          promise.push(_call('xo.getAllObjects', { filter: { id: id } }))
+        )
+        let isLocal = true
+        Promise.all(promise).then(response => {
```

Comment:
To avoid multiple loops witch impact the performance. I suggest this code:

```js
      const srPool = getObject(store.getState(), sr).$pool
      const isRemoteCopy = _vms.some(
        id => getObject(store.getState(), id).$pool !== srPool
      )
      if (isRemoteCopy) {
        error(_('copyVm'), _('upgradeNeeded'))
        return
      }
```